### PR TITLE
Refactor interactions

### PR DESCRIPTION
**Pull request by russellmcc**

### CODE REVIEW

**Comment by russellmcc** [comment]
Also snuck in some bugfixes

File: duneclient/src/actions.js [hunk 1]
```diff
@@ -74,35 +70,43 @@ const ActionArgs = ({me, state, args, setArgs, sendCommand, actionName, argSpec,
...
             <button disabled={interaction.mode != null} onClick={()=>{
-                const fixedArgs = args.split(" ").map((arg) => {
-                    if (arg.startsWith("$")) {
-                        return interaction[arg.slice(13)];
-                    }
-                    return arg;
-                }).join(" ");
-                sendCommand(`${actionName} ${fixedArgs}`);
-                setInteraction({mode: null});
+                sendCommand(`${actionName} ${[args].flat(Infinity).join(" ")}`);
```
Comment (eflynch) [inline duneclient/src/actions.js:81]: sick

File: duneclient/src/widget.js [hunk 2]
```diff
@@ -236,44 +240,30 @@ const Constant = ({value, setArgs}) => {
...
+        setArgs(newArgs);
+    };
+
+    let pieces = [
+            <div key="select" className={"select-on-map" + (mode === interaction.mode ? " on" : " off")} onClick={(e)=>{
+                setAndSelectArgs("");
+                setInteraction({
+                        mode,
+                        action: (val) => {
+                            setAndSelectArgs(val);
```
Comment (eflynch) [inline duneclient/src/widget.js:255]: oh yeah, nice

File: duneclient/src/session.js [hunk 3]
```diff
@@ -95,42 +95,87 @@ const Decks = ({state}) => {
...
+                return update(combinedState, {
+                    selection: {
+                        $unset: [mode]
+                    }
+                });
+            });
+        }
+    };
+
+    const wrapInteraction = (interaction, combinedState) => {
```
Comment (eflynch) [inline duneclient/src/session.js:145]: This looks like it was a little harry to get right

**Pull request merged**